src/librbd/internal.cc

```diff
@@ -444,6 +444,30 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
       RWLock::WLocker l(ictx->snap_lock);
       ictx->object_map.rollback(snap_id);
     }
+
+    {
+      RWLock::WLocker snap_locker(ictx->snap_lock);
+      RWLock::WLocker parent_locker(ictx->parent_lock);
+      std::map<librados::snap_t, SnapInfo>::const_iterator it = ictx->snap_info.find(snap_id);
+      if (it != ictx->snap_info.end()) {
+          const SnapInfo& si(it->second);
+          // current parent does not match the one of the snapshot being reverting to
+          bool should_update_parent = ictx->parent_md.spec != si.parent.spec ||
+                  ictx->parent_md.overlap != si.parent.overlap;
```

Comment:
If the overlap changes (but parent record still exists), the cls `set_parent` method will return `-EEXISTS` when trying to update the parent record.  Therefore, you will need to `remove_parent` first.  Also need to support the case where HEAD is associated with a parent but the rollback snapshot doesn't have a parent.